ocaml/xenopsd/scripts/swtpm-wrapper

```diff
@@ -85,38 +84,80 @@ def prepare_exec():
     sys.stdout.flush()
     sys.stderr.flush()
 
+def make_socket(fname):
+
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind(fname)
+    sock.listen(1)
+
+    return sock
+
+def check_state_needs_init(fname):
+
+    if not os.path.exists(fname):
+        return True
+
+    mode = os.stat(fname).st_mode
+
+    if not stat.S_ISBLK(mode):
+        return False
+
+    # Check if block device has non-zero header
+    with open(fname, "r") as file:
+        hdr = file.read(8)
+
+    if len(hdr) == 8 and hdr != "\0\0\0\0\0\0\0\0":
```

Comment:
It would be better to see if there is a magic number you can use here instead.

The source code defines `SWTPM_NVSTORE_LINEAR_MAGIC` - does that work?

Reply:
I based this code on what it does to 'delete' the tpm.  It always zeros it.
I only wanted the equivelant of 'if the file exists' which we had before.
I suppose the magic number would work too, I didnt try.